src/runtime/components/ChatMessages.vue

```diff
@@ -57,21 +57,21 @@ export interface ChatMessagesProps {
   ui?: ChatMessages['slots']
 }
 
-type ExtendSlotWithVersion<K extends keyof ChatMessageSlots>
+type ExtendSlotWithVersion<K extends keyof ChatMessageSlots, METADATA = unknown, DATA_PARTS extends UIDataTypes = UIDataTypes, TOOLS extends UITools = UITools>
   = ChatMessageSlots[K] extends (props: infer P) => any
-    ? (props: P & { message: UIMessage }) => any
+    ? (props: P & { message: UIMessage<METADATA, DATA_PARTS, TOOLS> }) => any
```

Comment:
```suggestion
type ExtendSlotWithVersion<K extends keyof ChatMessageSlots, T extends UIMessage[] = UIMessage[]>
  = ChatMessageSlots[K] extends (props: infer P) => any
    ? (props: P & { message: T[number] }) => any
```

This will create a union of all the available messages with their respective metadatas, types and tools